From: Huang Ying <ying.huang@intel.com>
Subject: resource: remove dependency on SPARSEMEM from GET_FREE_REGION
Date: Tue, 15 Oct 2024 13:15:54 +0800

We want to use the functions configured via GET_FREE_REGION in resource
kunit tests.  However, GET_FREE_REGION depends on SPARSEMEM.  This was
done to avoid a small amount of bloat in kernel/resource.c.  This means
resource kunit tests cannot be built on some architectures lacking
SPARSEMEM.  In fact, these functions doesn't depend on SPARSEMEM now.  So,
remove dependency on SPARSEMEM from GET_FREE_REGION.

Link: https://lore.kernel.org/lkml/20240922225041.603186-1-linux@roeck-us.net/
Link: https://lkml.kernel.org/r/20241015051554.294734-1-ying.huang@intel.com
Signed-off-by: "Huang, Ying" <ying.huang@intel.com>
Tested-by: Guenter Roeck <linux@roeck-us.net>
Cc: Nathan Chancellor <nathan@kernel.org>
Cc: Arnd Bergmann <arnd@arndb.de>
Cc: Dan Williams <dan.j.williams@intel.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Jonathan Cameron <jonathan.cameron@huawei.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/Kconfig |    1 -
 1 file changed, 1 deletion(-)

--- a/mm/Kconfig~resource-remove-dependency-on-sparsemem-from-get_free_region
+++ a/mm/Kconfig
@@ -1085,7 +1085,6 @@ config HMM_MIRROR
 	depends on MMU
 
 config GET_FREE_REGION
-	depends on SPARSEMEM
 	bool
 
 config DEVICE_PRIVATE
_
